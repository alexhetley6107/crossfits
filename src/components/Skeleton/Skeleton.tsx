import React from 'react';
import styled from 'styled-components';
import ContentLoader from 'react-content-loader';

export type SkeletonProps = {
  className?: string;
  width?: number;
  height?: number;
  radius?: number;
};

const SkeletonWrapper = styled.div<{ $height?: number }>`
  height: ${({ $height }) => $height}px;
`;

export const Skeleton: React.FC<SkeletonProps> = ({ width, height, radius = 6, className }) => {
  return (
    <SkeletonWrapper data-testid="skeleton" className={className} $height={height}>
      <ContentLoader
        speed={1}
        width={width}
        height={height}
        viewBox={`0 0 ${width} ${height}`}
        backgroundColor="#f3f3f3"
        foregroundColor="#ecebeb"
        style={{ borderRadius: radius + `px` }}
      >
        <rect x="0" y="0" width={width} height={height} />
      </ContentLoader>
    </SkeletonWrapper>
  );
};
