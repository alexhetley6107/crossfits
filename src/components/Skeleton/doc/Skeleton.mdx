import { Canvas, Meta } from '@storybook/blocks';
import { Example } from './Example.tsx';
import * as Skeleton from './Skeleton.stories.tsx';

<Meta of={Skeleton} title="Skeleton" />

# Skeleton

<Canvas of={Skeleton.Base} />

## Usage

```ts
import { Skeleton } from 'remindee-ui-kit';

export const Example = () => {
  return <Skeleton />;
};
```

### Arguments

- **className** - A string that sets css class to the root of component.
- **width** - A number that sets width of skeleton in `px`.
- **height** - A number that sets height of skeleton in `px`.
- **radius** - A number that sets border-radius of skeleton in `px`.
