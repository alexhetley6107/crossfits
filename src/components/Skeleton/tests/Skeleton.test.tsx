import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Skeleton } from '../Skeleton';
import '@testing-library/jest-dom';
import 'jest-styled-components';

describe('Skeleton component', () => {
  describe('General', () => {
    it('Render', () => {
      render(<Skeleton width={30} height={30} />);
      const el = screen.getByTestId('skeleton');
      expect(el).toBeInTheDocument();
    });
  });
});
