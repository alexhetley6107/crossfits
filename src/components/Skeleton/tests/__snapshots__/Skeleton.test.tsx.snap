// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton component > General > Dimensions 1`] = `
.c0 {
  height: 30px;
}

<div
  class="c0"
  data-testid="skeleton"
>
  <svg
    aria-labelledby="yh6816t-aria"
    height="30"
    role="img"
    style="border-radius: 6px;"
    viewBox="0 0 30 30"
    width="30"
  >
    <title
      id="yh6816t-aria"
    >
      Loading...
    </title>
    <rect
      clip-path="url(#yh6816t-diff)"
      height="100%"
      role="presentation"
      style="fill: url(#yh6816t-animated-diff);"
      width="100%"
      x="0"
      y="0"
    />
    <defs>
      <clippath
        id="yh6816t-diff"
      >
        <rect
          height="30"
          width="30"
          x="0"
          y="0"
        />
      </clippath>
      <lineargradient
        gradientTransform="translate(-2 0)"
        id="yh6816t-animated-diff"
      >
        <stop
          offset="0%"
          stop-color="#f3f3f3"
          stop-opacity="1"
        />
        <stop
          offset="50%"
          stop-color="#ecebeb"
          stop-opacity="1"
        />
        <stop
          offset="100%"
          stop-color="#f3f3f3"
          stop-opacity="1"
        />
        <animatetransform
          attributeName="gradientTransform"
          dur="1s"
          repeatCount="indefinite"
          type="translate"
          values="-2 0; 0 0; 2 0"
        />
      </lineargradient>
    </defs>
  </svg>
</div>
`;

exports[`Skeleton component > General > Render 1`] = `
.c0 {
  height: 30px;
}

<div
  class="c0"
  data-testid="skeleton"
>
  <svg
    aria-labelledby="5rmbwt-aria"
    height="30"
    role="img"
    style="border-radius: 20px;"
    viewBox="0 0 30 30"
    width="30"
  >
    <title
      id="5rmbwt-aria"
    >
      Loading...
    </title>
    <rect
      clip-path="url(#5rmbwt-diff)"
      height="100%"
      role="presentation"
      style="fill: url(#5rmbwt-animated-diff);"
      width="100%"
      x="0"
      y="0"
    />
    <defs>
      <clippath
        id="5rmbwt-diff"
      >
        <rect
          height="30"
          width="30"
          x="0"
          y="0"
        />
      </clippath>
      <lineargradient
        gradientTransform="translate(-2 0)"
        id="5rmbwt-animated-diff"
      >
        <stop
          offset="0%"
          stop-color="#f3f3f3"
          stop-opacity="1"
        />
        <stop
          offset="50%"
          stop-color="#ecebeb"
          stop-opacity="1"
        />
        <stop
          offset="100%"
          stop-color="#f3f3f3"
          stop-opacity="1"
        />
        <animatetransform
          attributeName="gradientTransform"
          dur="1s"
          repeatCount="indefinite"
          type="translate"
          values="-2 0; 0 0; 2 0"
        />
      </lineargradient>
    </defs>
  </svg>
</div>
`;
